backend/src/main/kotlin/app/ehrenamtskarte/backend/verification/database/repos/CardRepository.kt

```diff
@@ -8,11 +8,20 @@ import app.ehrenamtskarte.backend.verification.database.CardEntity
 import app.ehrenamtskarte.backend.verification.database.Cards
 import app.ehrenamtskarte.backend.verification.database.CodeType
 import org.jetbrains.exposed.dao.id.EntityID
+import org.jetbrains.exposed.sql.SqlExpressionBuilder.eq
+import org.jetbrains.exposed.sql.SqlExpressionBuilder.inList
+import org.jetbrains.exposed.sql.SqlExpressionBuilder.isNull
 import org.jetbrains.exposed.sql.and
+import org.jetbrains.exposed.sql.deleteWhere
 import org.jetbrains.exposed.sql.select
 import java.time.Instant
 
 object CardRepository {
+    fun deleteInactiveCardsByHash(project: String, regionId: Int, cardInfoHashList: List<ByteArray>) {
+        Cards.deleteWhere {
+            firstActivationDate.isNull() and (cardInfoHash inList cardInfoHashList) and (Cards.regionId eq regionId)
+        }
```

Comment:
Do you think we should return an error if some cards were not found or were already active or have the wrong regionId?

Reply:
Not sure. For now this should be sufficient, as it's just the fallback for when pdf generation fails. But if we want to implement deletion by the user or an administrator this could look different. @f1sh1918 what do you think?

Reply:
If we leave it like this, we should probably document the API endpoint that it silently fails / ignores these cases.

Reply:
i think users shouldn't be able to delete cards but revoke them if we gonna implement a "revoke" function one day.
Btw how can i test a pdf creation failure to initiate the deletion @sarahsporck ?

Reply:
I just added a throw Error in the PdfFactory to test this 👍🏼